subsys/bluetooth/audio/tbs_client.c

```diff
@@ -600,6 +599,83 @@ static uint8_t notify_handler(struct bt_conn *conn,
 	return BT_GATT_ITER_CONTINUE;
 }
 
+/* Common function to read tbs_client strings which may require long reads */
+static uint8_t handle_string_long_read(struct bt_conn *conn, uint8_t err,
+				       struct bt_gatt_read_params *params,
+				       const void *data,
+				       uint16_t length,
+				       bt_tbs_client_read_string_cb cb,
+				       bool truncate_silently)
+{
+	struct bt_tbs_instance *inst = CONTAINER_OF(params,
+						    struct bt_tbs_instance,
+						    read_params);
+	uint16_t offset = params->single.offset;
+	uint8_t inst_index = tbs_index(conn, inst);
+	const char *received_string;
+	int tbs_err = err;
+
+	if ((tbs_err == 0) && (data != NULL) &&
+	    (net_buf_simple_tailroom(&inst->net_buf) < length)) {
+		if (truncate_silently == true) {
+			/* Use the remaining buffer and continue reading */
+			length = net_buf_simple_tailroom(&inst->net_buf);
```

Comment:
We may want to log (with debug logging) when we are truncating the string. 

In which case, `truncate_silently` may not be the best name for the variable; perhaps just `truncate` or `truncatable`?

Reply:
I will add log. 
truncatable is a better name.